plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java

```diff
@@ -0,0 +1,331 @@
+            return WriteMapping.sliceMapping("varbinary", varbinaryWriteFunction());
+        }
+        if (type == DATE) {
+            return WriteMapping.longMapping("date", dateWriteFunctionUsingSqlDate());
+        }
+
+        if (type instanceof TimeType) {
+            TimeType timeType = (TimeType) type;
+            if (timeType.getPrecision() <= SNOWFLAKE_MAX_SUPPORTED_TIMESTAMP_PRECISION) {
+                return WriteMapping.longMapping(format("time(%s)", timeType.getPrecision()), timeWriteFunction(timeType.getPrecision()));
+            }
+            return WriteMapping.longMapping(format("time(%s)", SNOWFLAKE_MAX_SUPPORTED_TIMESTAMP_PRECISION), timeWriteFunction(SNOWFLAKE_MAX_SUPPORTED_TIMESTAMP_PRECISION));
+        }
+
+        if (type == TIMESTAMP_NANOS) {
+            return WriteMapping.longMapping("timestamp", timestampWriteFunction(TIMESTAMP_NANOS));
+        }
+
+        throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName());
+    }
+
+    @Override
+    public Optional<JdbcExpression> implementAggregation(ConnectorSession session, AggregateFunction aggregate, Map<String, ColumnHandle> assignments)
+    {
+        return aggregateFunctionRewriter.rewrite(session, aggregate, assignments);
+    }
+
+    private static Optional<JdbcTypeHandle> toTypeHandle(DecimalType decimalType)
+    {
+        return Optional.of(new JdbcTypeHandle(Types.NUMERIC, Optional.of("decimal"), Optional.of(decimalType.getPrecision()), Optional.of(decimalType.getScale()), Optional.empty(), Optional.empty()));
+    }
+
+    @Override
+    protected Optional<BiFunction<String, Long, String>> limitFunction()
+    {
+        return Optional.of((sql, limit) -> sql + " LIMIT " + limit);
+    }
+
+    @Override
+    public boolean isLimitGuaranteed(ConnectorSession session)
+    {
+        return true;
+    }
+
+    private ColumnMapping jsonColumnMapping()
+    {
+        return ColumnMapping.sliceMapping(
+                jsonType,
+                (resultSet, columnIndex) -> jsonParse(utf8Slice(resultSet.getString(columnIndex))),
+                varcharWriteFunction(),
+                DISABLE_PUSHDOWN);
+    }
+
+    private static Optional<ColumnMapping> getUnsignedMapping(JdbcTypeHandle typeHandle)
+    {
+        if (!typeHandle.getJdbcTypeName().isPresent()) {
+            return Optional.empty();
+        }
+
+        String typeName = typeHandle.getJdbcTypeName().get();
+        if (typeName.equalsIgnoreCase("tinyint unsigned")) {
+            return Optional.of(smallintColumnMapping());
+        }
+        if (typeName.equalsIgnoreCase("smallint unsigned")) {
+            return Optional.of(integerColumnMapping());
+        }
+        if (typeName.equalsIgnoreCase("int unsigned")) {
+            return Optional.of(bigintColumnMapping());
+        }
+        if (typeName.equalsIgnoreCase("bigint unsigned")) {
+            return Optional.of(decimalColumnMapping(createDecimalType(20)));
+        }
```

Comment:
Does Snowflake support unsigned types? Remove or add test cases to `TestSnowflakeTypeMapping`. 

https://github.com/trinodb/trino/pull/10387#discussion_r780935822 means type mapping should fail if we remove condition in `toColumnMapping` or `toWriteMapping` in `SnowflakeClient`.

Reply:
No relevant descriptions of snowflake's unsigned type data were found. I don't think it's supported. So I removed it.